### ISSUE TITLE
Max size of head/tail

### ISSUE THREAD

**Issue by ongkoonhan** [jnk0le/Ring-Buffer#25]
I noticed the implementation only increments head/tail atomic size_t variables.
If **head/tail** reaches the **max value of size_t** and overflows, does the implementation still perform as expected?

**Comment by jnk0le** [comment]
yes, as the two's complement numbers add/subtract correctly across unsigned overflow.
